Calendar/AuthenticationHelper.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using System.Threading.Tasks;
+using System.Configuration;
+using System.Net.Http.Headers;
+using System.Diagnostics;
+
+namespace Calendar
+{
+    public class AuthenticationHelper
+    {
+        // The Client ID is used by the application to uniquely identify itself to the v2.0 authentication endpoint.
+        static string clientId = ConfigurationManager.AppSettings["clientId"].ToString();
+        public static string[] Scopes = { "Calendars.Read","Calendars.ReadWrite" };
+
+        public static PublicClientApplication IdentityClientApp = new PublicClientApplication(clientId);
+
+        private static GraphServiceClient graphClient = null;
+
+        // Get an access token for the given context and resourceId. An attempt is first made to
+        // acquire the token silently. If that fails, then we try to acquire the token by prompting the user.
+        public static GraphServiceClient GetAuthenticatedClient()
```

Comment:
This is copied from old samples that don't use the AuthHandler middleware for getting a token.  Please use the pattern shown here https://github.com/microsoftgraph/dotnetcore-console-sample/blob/master/base-console-app/Program.cs#L54

Reply:
Ping @titusgicheru

Reply:
@darrelmiller @MIchaelMainer  can I convert this project to a .NET Core and implement the above

Reply:
@darrelmiller Why would we create a custom IAuthenticationProvider instead of using one of our off the shelf ones from Microsoft.Graph.Auth? Is the intention to provide an example of how to do it?

Reply:
Please use the provider in Microsoft.Graph.Auth. Ideally we make this .Net Core as well.